docs/specs/auth-plugin.md

```diff
@@ -0,0 +1,73 @@
+# Authorization Plugin
+
+Add authorization and authentication to Koop services. Dive into the docs below or check out a working example [here](https://github.com/koopjs/koop-auth-direct-file).
+
+## Index.js
+Each authorization plugin must have a file called `index.js`.  Its purpose is to return a registration object that defines the type of Koop module and provides a set of configured functions that Koop will attach to each provider's Model prototype.  For example, the koop-auth-file-direct plugin exports a function that configures and returns a registration object that includes three functions, `getAuthenticateSpecification`, `authenticate`, and `authorize`.
+<script src="https://gist.github.com/rgwozdz/4098e0e18cc85299f4444a3e64a7c68b.js"></script>
+
+
+### Function: getAuthenticationSpecification
+
+Authorization plugins are required to return a function called "getAuthenticationSpecification".  Its purpose is to configure a function that is used to describe the authentication specification. It should have the following signature:
```

Comment:
is this function still required after all the changes we've made? is it possible to use defaults instead?

Reply:
This is a good question. We still need an `authenticationSpecification`  because output services needs the provider name and `useHttp` option to construct the `tokenServicesUrl`. 
But I think this could be simplified.  No real need to pass in the provider name and get back a function.  There should be a simple function that returns an object with `useHttp` (and anything else that gets added as options in the future).  Koop-core could then decorate the object with the provider's name before attaching the authenticationSpecification object to the Model.

Reply:
I was able to simplify, but it takes three PRs because of the coupling between the Auth Plugin, koop-core, and koop-output-geoservices:

https://github.com/koopjs/koop-auth-direct-file/pull/3
https://github.com/koopjs/koop-core/pull/36
https://github.com/koopjs/koop-output-geoservices/pull/16